cgi-bin/CATS/UI/Snippets.pm

```diff
@@ -52,6 +52,57 @@ sub edit_save {
     $form->edit_save($p) and msg(1075, $p->{name});
 }
 
+sub problem_visible {
+    my ($cpid) = @_;
+    return 1 if $is_root;
+
+    my $c = $dbh->selectrow_hashref(qq~
+        SELECT
+            CAST(CURRENT_TIMESTAMP - $CATS::Time::contest_start_offset_sql AS DOUBLE PRECISION) AS since_start,
+            C.local_only, C.id AS orig_cid, C.show_packages, C.is_hidden,
+            CA.id AS caid, CA.is_jury, CA.is_remote, CA.is_ooc
+            FROM contests C INNER JOIN contest_problems CP ON C.id = CP.contest_id
```

Comment:
1) Unindent this and following lines
2) Swap `contest_problems` and `contest`, just for readability